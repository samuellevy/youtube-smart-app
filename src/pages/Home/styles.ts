import styled from 'styled-components';
import { colors } from '../../global';

export const Container = styled.div`
  padding: 25px 0 0 25px;
  padding-left: 250px;
`;
export const PageTitle = styled.h1`
  color: ${colors.gray};
  font-size: 1vw;
  text-transform: uppercase;
`;
export const Track = styled.div`
  display: flex;
`;
export const CategoryTitle = styled.h1`
  width: 100%;
`;
