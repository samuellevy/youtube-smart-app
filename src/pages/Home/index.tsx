import React, { useEffect, useState } from 'react';
import Track from '../../components/Track';
import { useKeyboardContext, changeComponent } from '../../context/KeyboardContext';
// import VideoCard from '../../components/VideoCard';

import * as S from './styles';

const Home: React.FC = () => {
  const { keyboard, dispatch }: any = useKeyboardContext();
  const [activePage, setActivePage] = useState(false);
  const [activeItem, setActiveItem] = useState(0);

  const tracks = [
    { title: 'MÃºsica', id: 10 },
    { title: 'Gaming', id: 20 },
  ];

  useEffect(() => {
    dispatch(changeComponent('menu'));
  }, []);

  useEffect(() => {
    if (keyboard.component === 'home') {
      setActivePage(true);
      controlHandler(keyboard.key);
    }
  }, [keyboard]);

  const controlHandler = (key: string) => {
    let newActiveItem = activeItem;
    switch (key) {
      case 'ArrowLeft':
        dispatch(changeComponent('menu'));
        setActivePage(false);
        break;
      case 'ArrowDown':
        newActiveItem += tracks.length - 1 > newActiveItem ? 1 : 0;
        break;
      case 'ArrowUp':
        newActiveItem -= newActiveItem === 0 ? 0 : 1;
        break;
      default:
        break;
    }
    setActiveItem(newActiveItem);
  };

  return (
    <S.Container active={activePage}>
      <S.PageTitle>Home</S.PageTitle>
      {tracks.map((item, key) => (
        <Track active={key === activeItem} title={item.title} id={item.id} />
      ))}
    </S.Container>
  );
};
export default Home;
