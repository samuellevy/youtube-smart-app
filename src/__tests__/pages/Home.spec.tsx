import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Home from '../../pages/Home';

Enzyme.configure({ adapter: new Adapter() });

describe('<Home /> component', () => {
  it('should render properly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
