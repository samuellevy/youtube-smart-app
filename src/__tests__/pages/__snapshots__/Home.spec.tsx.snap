// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> component should render properly 1`] = `
<div
  className="sc-jrAGrp fiSZBI"
  data-testid="home"
>
  <h1
    className="sc-kEjbxe gHFyGX"
  >
    HOME
  </h1>
</div>
`;
